### PR TITLE
[feature][broker] offloader the data by topic compacted

### PR DESCRIPTION
**Pull request by azhsmesos**
ospp 开源活动 任务

### CODE REVIEW

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/NullLedgerOffloader.java [hunk 1]
```diff
@@ -45,6 +47,11 @@ public CompletableFuture<Void> offload(ReadHandle ledger,
         return promise;
     }
 
+    @Override
+    public CompletableFuture<Void> offload(ByteBuf buf, long ledgerId, String topicName, UUID uid) {
+        return null;
```
Comment (Technoboy-) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/NullLedgerOffloader.java:52]: Should return `CompletableFuture.completedFuture(null);`

File: managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/OffloadLedgerDeleteTest.java [hunk 2]
```diff
@@ -110,6 +112,11 @@ public CompletableFuture<Void> offload(ReadHandle ledger,
             return promise;
         }
 
+        @Override
+        public CompletableFuture<Void> offload(ByteBuf buf, long ledgerId, String topicName, UUID uid) {
+            return null;
```
Comment (Technoboy-) [inline managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/OffloadLedgerDeleteTest.java:117]: CompletableFuture.completedFuture(null);

File: managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/OffloadPrefixReadTest.java [hunk 3]
```diff
@@ -231,6 +235,11 @@ public CompletableFuture<Void> offload(ReadHandle ledger,
             return promise;
         }
 
+        @Override
+        public CompletableFuture<Void> offload(ByteBuf buf, long ledgerId, String topicName, UUID uid) {
+            return null;
```
Comment (Technoboy-) [inline managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/OffloadPrefixReadTest.java:240]: CompletableFuture.completedFuture(null);

**Comment by codelipenghui** [comment]
We need to start with a proposal for this one since it's a new feature introduced to Pulsar.

**Pull request closed without merging**